x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -64,10 +68,27 @@ fragment EXPONENT
     : [Ee] [+-]? DIGIT+
     ;
 
+fragment ASPERAND
+    : '@'
+    ;
+
+fragment BACKQUOTE
+    : '`'
+    ;
+
+fragment BACKQUOTE_BLOCK
+    : ~'`'
+    | '``'
+    ;
+
 fragment UNDERSCORE
     : '_'
     ;
 
+fragment UNQUOTED_ID_BODY
+    : (LETTER | DIGIT | UNDERSCORE)
```

Comment:
++ to removing the dot again from the unquoted identifier. I agree that dots will likely need to retain special meaning to navigate nested objects. If I understand correctly, this should also bring back the ability to quote just part of an identifier, e.g. 
```
foo.bar.`1234asdf`
```